edx_exams/apps/core/tests/test_api.py

```diff
@@ -289,6 +290,92 @@ def test_with_no_attempt(self):
         self.assertIsNone(get_attempt_by_id(111111111))
 
 
+@ddt.ddt
+class TestGetLatestAttemptForUser(ExamsAPITestCase):
+    """
+    Test for the API utility function `get_latest_attempt_for_user`
+    """
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = '11111111'
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+        self.user = User.objects.create(
+            id=3,
+            username="jerry",
+            email="jerry@example.com",
+            lms_user_id=2
+        )
+
+    def create_mock_exam_attempt(self, user, attempt_number, status, start_time):
+        return ExamAttempt.objects.create(
+            user=user,
+            exam=self.exam,
+            attempt_number=attempt_number,
+            status=status,
+            start_time=start_time,
+            allowed_time_limit_mins=None
+        )
+
+    def test_get_latest_exam_attempt_for_user(self):
+        """
+        Test that the GET function in the ExamAttempt view returns
+        the latest exam attempt for a user
+        """
+
+        one_hour_ago = datetime.now() - timedelta(hours=1)
+        expected_attempt = self.create_mock_exam_attempt(self.user, 1, ExamAttemptStatus.started, datetime.now())
+        self.create_mock_exam_attempt(self.user, 2, ExamAttemptStatus.started, one_hour_ago)
+        self.create_mock_exam_attempt(self.user, 3, ExamAttemptStatus.ready_to_submit, one_hour_ago)
+        self.create_mock_exam_attempt(self.user, 4, ExamAttemptStatus.created, None)
+        self.create_mock_exam_attempt(self.user, 5, ExamAttemptStatus.submitted, None)
+        self.create_mock_exam_attempt(self.user, 6, ExamAttemptStatus.rejected, None)
```

Comment:
I feel like this is testing multiple assertions. You probably only need the first two created attempts, as opposed to 6, to test. 

Your other test covers the case where there is not a start time